Framework/Algorithms/test/RebinTest.h

```diff
@@ -146,7 +199,139 @@ class RebinTest : public CxxTest::TestSuite {
     bool dist = rebindata->isDistribution();
     TS_ASSERT(dist);
 
-    TS_ASSERT(checkBinWidthMonotonic(rebindata, false));
+    TS_ASSERT(checkBinWidthMonotonic(rebindata, Monotonic::INCREASE));
+
+    AnalysisDataService::Instance().remove("test_in1D");
+    AnalysisDataService::Instance().remove("test_out");
+  }
+
+  // Check that logarithmic binning is performed if the BinningMode is set, even if positive binwidth
+  // Run once in the default manner, then run again with BinningMode set, and compare.
+  void testworkspace1D_logarithmic_binning_modeset() {
+    Workspace2D_sptr test_in1D = Create1DWorkspace(50);
+    test_in1D->setDistribution(true);
+    AnalysisDataService::Instance().add("test_in1D", test_in1D);
+
+    Rebin rebin_default;
+    rebin_default.initialize();
+    rebin_default.setPropertyValue("InputWorkspace", "test_in1D");
+    rebin_default.setPropertyValue("OutputWorkspace", "test_out_default");
+    rebin_default.setPropertyValue("Params", "1.0,-1.0,1000.0");
+    TS_ASSERT_THROWS_NOTHING(rebin_default.execute());
+    TS_ASSERT(rebin_default.isExecuted());
+    MatrixWorkspace_sptr rebindata_default =
+        AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>("test_out_default");
+    auto &outX_default = rebindata_default->x(0);
+
+    // now run again but with positive step, using logarithmic mode
+    Rebin rebin_again;
+    rebin_again.initialize();
+    rebin_again.setPropertyValue("InputWorkspace", "test_in1D");
+    rebin_again.setPropertyValue("OutputWorkspace", "test_out_again");
+    rebin_again.setPropertyValue("Params", "1.0,1.0,1000.0");
+    rebin_again.setPropertyValue("Power", "0.5");
```

Comment:
should the testing of the power being set to zero have its own test?

Reply:
I thought that it did.  If it does not, then you are correct, that should be a specific test.